openapi.yml

```diff
@@ -134,9 +134,13 @@ components:
           description: Constant identifier for Recommendation type objects.
           enum:
             - Recommendation
+        id:
+          type: string
+          description: String identifier for the Recommendation. This value is expected to be different on each request.
```

Comment:
So nothing is wrong with it from a breaking standpoint.

However the name `id` vs `recommendationId` is hard to comment on without some help understanding it.

Is it the idea that we migrate the client from using `tileId` to this new id? If so, I think `id` is right.

Right now we convert `tileId` to `id` in the client anyway because of backwards comp reasons while the client was supporting both apis. Having it go back to just `id` is nice and clean.

If it's a new id, is different from `tileId`, then it would be weird having 2 different ids, both being called at one point `id`. If that makes sense.

Reply:
Thanks, that context is helpful.

In the [BugZilla meta ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1854245) I proposed adding both this `id` and `tileId` to Glean, but based on what you said, it sounds better only to emit `id` to Glean. I believe that would work for us. I'll update the BugZilla ticket accordingly.

Reply:
@ScottDowne will check that `id` will is not used by any supported version of Firefox Desktop. I'll add a tag to hold deployment until that's done.

Reply:
The browser does not use `id` for any versions that use this API. You can see it here: https://searchfox.org/mozilla-central/source/browser/components/newtab/lib/DiscoveryStreamFeed.jsm#1465-1473

Older versions use the fallback legacy api we setup, which I imagine is the same, so we should be fine here.